usefulstuff.c

```diff
@@ -53,7 +53,7 @@ ZEND_EXTERN_MODULE_GLOBALS(xdebug)
 
 #define READ_BUFFER_SIZE 128
 
-char* xdebug_fd_read_line_delim(int socketfd, fd_buf *context, int type, unsigned char delim, int *length)
+int xdebug_fd_read_line_delim(int socketfd, fd_buf *context, int type, unsigned char delim, char **cmd, int *length, int block)
```

Comment:
I prefer having the IN parameters (such as the new `int block`) before the OUT parameters (such as `char **cmd`, and `int *length`).